### PR TITLE
Add documentation with Material Mkdocs and GitHub Action to build/deploy

### PR DESCRIPTION
**Pull request by plusmobileapps**
# Overview 

Add documentation with [Material MkDocs](https://squidfunk.github.io/mkdocs-material/) to the repository. 

## Description 

* ported the necessary files from Material MkDocs into the repository with a skeleton configuration for Decompose
* added a GitHub action that can automatically build and deploy documentation changes to GitHub Pages. So the website after this should be  https://arkivanov.github.io/Decompose

## Demo 

https://plusmobileapps.github.io/Decompose

## Issue

[Github Issue](https://github.com/arkivanov/Decompose/issues/10)

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - run: pip install mkdocs-material
+      - run: pip install mkdocs-minify-plugin
+      - run: mkdocs gh-deploy --force
```
Comment (arkivanov) [inline .github/workflows/documentation.yml:17]: Where does it deploy?

Comment (plusmobileapps) [reply .github/workflows/documentation.yml:17]: By default it deploys to the gh-pages branch on GitHub. Although there is an argument that can be added if you wish to change the branch this pushes to.

https://www.mkdocs.org/user-guide/deploying-your-docs/

**Comment by arkivanov** [comment]
Thanks! I have few more questions. 
Can we/should we deploy to the main branch? 
Why the deployment is a thing here? E.g. I have setup docs for MVIKotlin library [here](https://arkivanov.github.io/MVIKotlin/). So I just selected main branch, a folder and a theme, and that's it. All MD files are in the `docs` folder. There is `index.md`, which is the main page. The site is updated automatically when I change docs files. No workflows required.

**Comment by plusmobileapps** [comment]
So after looking between the two types of documentation, the MVIKotlin library uses one of the preconfigured GH pages themes that will automatically build a Jekyll site and deploy changes from the `main` branch. 

Material Mkdocs being a custom theme for GitHub Pages, will have a similar setup having the `main` branch with a `docs/` folder and `mkdocs.yml` file. Then in order to translate  Material Mkdocs into a Jekyll site for GitHub Pages, is where the deployment script comes in which will run the `mkdocs gh-deploy --force` command pushing the built Jekyll site to the `gh-pages` branch. So the repository settings would need to be configured to have the GitHub Pages site built from the `gh-pages` branch. 

Just two different ways for building a documentation website on GitHub Pages.  I'm fine using one of the themes provided as the setup is simpler only needing to edit the markdown files for preview. I just like the feature set and UI Material Mkdocs brings to documentation, that I think its worth the extra setup of seeing the changes locally in the browser before deploying with docker and the GitHub Action script to automatically deploy the changes to your GitHub Pages site.

**Comment by arkivanov** [comment]
Thanks for the explanation! Let's try MkDocs and see how it goes. Should I first configure the repository? If so, should I do it before merging this branch?

**Comment by plusmobileapps** [comment]
I believe you will need to merge the branch first so that the GitHub Action will build and deploy to the `gh-pages` branch. Then you should be able to go to the repository settings and select that branch as the publishing source for GitHub Pages. This is how I configured the repository on my fork to get the documentation up and running. 

Screen Shot 2021-01-18 at 12 01 50

**Pull request merged**